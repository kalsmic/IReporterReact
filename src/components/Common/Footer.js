import React from 'react';


const Footer = () => (

  <div className="wrapper__footer">
    Ireporter @copyright 2019
  </div>

);
export default Footer;
