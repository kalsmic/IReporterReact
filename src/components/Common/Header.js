import React from 'react';


const Header = () => (

  <div className="wrapper__header">
    Ireporter
  </div>

);
export default Header;
