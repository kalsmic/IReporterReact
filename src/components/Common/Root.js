import React, { Component } from 'react';
import PropTypes from 'prop-types';

import NavBar from '../NavBar';
import Header from './Header';
import Footer from './Footer';

class Root extends Component {
  render() {
    const { children } = this.props;

    return (
      <div className="wrapper">
        <Header />
        <div className="wrapper__menu border-round-lg">
          <NavBar />
        </div>
        <div className="wrapper__page-content border-round-lg">
          {children}
        </div>
        <Footer />
      </div>
    );
  }
}

Root.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Root;
