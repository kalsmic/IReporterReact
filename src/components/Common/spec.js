import { shallow } from 'enzyme';
import React from 'react';
import Footer from './Footer';
import Header from './Header';
import Root from './Root';

describe('CreateIncident Component', () => {
  const divComponent = <div>my Div</div>;
  const footer = shallow(<Footer />);
  const header = shallow(<Header />);
  const root = shallow(<Root>divComponent</Root>);


  it('should render without crushing', () => {
    expect(header).toMatchSnapshot();
    expect(footer).toMatchSnapshot();
    expect(root).toMatchSnapshot();
  });

  it('should wrap its children', () => {
    expect(root.text()).toBe('<Header /><NavBar />divComponent<Footer />');
  });
});
