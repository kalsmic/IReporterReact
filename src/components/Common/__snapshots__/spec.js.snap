// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIncident Component should render without crushing 1`] = `
<div
  className="wrapper__header border-round-lg"
>
  <b>
    Ireporter
  </b>
</div>
`;

exports[`CreateIncident Component should render without crushing 2`] = `
<div
  className="wrapper__footer border-round-lg"
>
  Ireporter Â© 2019
</div>
`;

exports[`CreateIncident Component should render without crushing 3`] = `
<div
  className="wrapper"
>
  <Header />
  <div
    className="wrapper__menu border-round-lg"
  >
    <NavBar />
  </div>
  <div
    className="wrapper__page-content border-round-lg"
  >
    divComponent
  </div>
  <Footer />
</div>
`;
